### DISCUSSION TITLE
关于 deepin 开源项目开发流程迁移至 GitHub 的说明

### DISCUSSION THREAD

**Discussion by BLumia** [linuxdeepin/developer-center#3050] [Announcements | 公告]
[English Version](https://github.com/linuxdeepin/developer-center/discussions/3051)

各位 deepin 社区的小伙伴们好！

正如[之前所提到的](https://github.com/linuxdeepin/developer-center/issues/2830)，我们正在改进与 deepin 社区相关的工作流程，以便使得我们的工作流程更加公开和透明，也可以使大家可以更方便的参与到 deepin 开源社区中。第一阶段我们计划将之前沉积的 GitHub Issue 进行验证和处理，现第一阶段的工作也已基本告一段落，我们接下来则将计划将所有的 deepin 社区开源项目的开发流程分批次逐步迁移回 GitHub，以便更方便的接受来自社区的 Pull Request。这个过程中，我们也会完善相应的文档以帮助社区能够更方便的了解参与流程，并参与进来。这篇公告将涉及我们与之相关的计划，包含我们已经将开发流程迁移至 GitHub 的项目列表，以及计划在近期进行迁移的项目清单。

## 项目迁移进度

*注：下面用 ✅ 标记的项为目前已完成开发流程至 GitHub 迁移的项目*

事实上，我们已经在二月和三月逐步迁移了一批项目，已经进行迁移的项目包括：

- ✅deepin-manual
- ✅deepin-music
- ✅deepin-camera
- ✅deepin-font-manager
- ✅deepin-gtk-theme
- ✅dtkcommon
- ✅dtkcore
- ✅dtkgui
- ✅dtkwidget
- ✅qt5integration
- ✅qt5platform-plugins

接下来，我们将计划继续把更多 deepin 的开源项目的开发流程逐步迁移至社区。其中，我们计划在四月优先迁移下列（非 DDE 桌面环境的）项目：

- ✅deepin-screen-recorder
- ✅deepin-calculator
- ✅deepin-terminal
- ✅dde-control-center
- ✅dde-file-manager
- ✅deepin-system-monitor

与此同时，我们也计划在四月份将尽可能多的 DDE 桌面环境组件对应的项目的开发流程迁移至 GitHub，目前计划包括下列项目：

- ✅dde-daemon
- ✅deepin-anything
- ✅dde-launcher
- ✅dde-dock
- ✅gio-qt
- ✅udisks2-qt5
- ⚠️dde-network-utils // 已废弃，请使用 networkmanager-qt
- ✅dde-network-core
- ✅dde-polkit-agent
- ✅deepin-face
- ✅seetaface-\*
- ✅dareader
- ✅dde-qt-dbus-factory
- ✅dde-session-ui
- ✅dde-session-shell
- ✅deepin-desktop-schemas
- ✅deepin-network-proxy
- ✅deepin-sound-theme
- ✅deepin-gettext-tools
- ✅dde-clipboard
- ✅startdde
- ⌛dde-grand-search
- ✅go-lib
- ✅go-x11-client
- ✅go-dbus-factory
- ✅go-gir-generator
- ✅dde-api
- ✅dde-wldpms
- ✅dde-wloutput-daemon
- ✅lastore-daemon
- ✅dde-kwin
- ✅deepin-kwin

以上项目均为我们计划在四月进行迁移的项目列表，我们将尽可能早的完成迁移，并不定期将迁移状态更新至此帖中。我们仍然有一些项目未出现在上述列表中，未列出的项目将会于上述列表迁移任务完成后继续迁移。如果您有对上述内容相关的任何疑问，都可以在此帖子下进行回复，或[在 discussion 讨论板中发起新的帖子](https://github.com/linuxdeepin/developer-center/discussions)与我们进行讨论反馈。我们会确保最终将所有 deepin 社区开源项目的开发流程迁移到 GitHub，以便更方便的与社区贡献者一同打造更好的 DDE/deepin！

**Comment by BLumia** [comment]
四月迁移已告一段落，五月项目迁移计划如下：

- ✅deepin-album
- ✅deepin-movie-reborn
- ✅deepin-image-viewer
- ✅deepin-compressor
- ✅deepin-deb-installer
- ✅deepin-draw
- ✅deepin-boot-maker
- ✅deepin-clone
- ✅deepin-voice-note
- ✅deepin-log-viewer
- ✅image-editor
- ✅deepin-ocr
- ✅dde-introduction
- ⌛deepin-diskmanager
- ⌛deepin-fcitxconfigtool-plugin
- ✅dde-wallpapers
- ✅deepin-wallpapers
- ✅deepin-plymouth
- ✅deepin-desktop-base
- ✅dde-account-faces
- ✅deepin-icon-theme
- ✅deepin-cursor-theme
- ✅default-settings
- ✅deepin-lianliankan
- ✅deepin-gomoku
- ✅deepin-downloader // 注：deepin-download 被此项目取代
- ✅dde-app-services

另外，仍有一些其它项目未涵盖在此列表中，其中一部分为不必要列出的配置文件、资源或工具类项目，另一部分为计划在此批迁移后计划尽快进行迁移的其它项目。您可以参阅[这里](https://github.com/linuxdeepin/.github/blob/master/organization.yaml#L4-L58)来获得一份包含杂碎项目在内的、已迁移到 GitHub 进行开发的完整项目清单。若您有其它问题或建议，也欢迎在讨论板发起新的主题进行讨论。感谢支持！

**Reply by felixonmars** [reply]
最后两个项目是否考虑用一个正常（和其他项目一致）的名字当仓库名？

**Reply by BLumia** [reply]
> 最后两个项目是否考虑用一个正常（和其他项目一致）的名字当仓库名？

目前也不止是这俩有点奇怪，下载器也是这种包名风格的仓库名，迁的时候会确认一下 = =

不是这种风格名称的仓库也有那种名字很怪的，比如 dareader 这看名字一头雾水的仓库，不过感觉这种问题得另外处理了...

**Reply by BLumia** [reply]
> 最后两个项目是否考虑用一个正常（和其他项目一致）的名字当仓库名？

Update: 连连看和五子棋的仓库名称已经改正常了。下载器的仓库名称随后迁移的时候我再确认下，应该也要改成正常一点的名字

**Reply by BLumia** [reply]
Update: 下载器会使用 deepin-downloader 作为仓库名，这样应该就没有其它使用反着的域名的格式的仓库名了

**Comment by BLumia** [comment]
根据目前情况确认，存在一个已完成迁移流程但实际仍在内部进行开发的项目：

- ✅dde-grand-search

五月迁移已告一段落，其中两个项目延迟至六月迁移：

- ✅deepin-diskmanager
- ✅deepin-fcitxconfigtool-plugin

六月项目迁移计划如下：

- ✅dde-wloutput
- ✅dde-calendar
- ⌛deepin-editor
- ⌛deepin-reader
- ⌛deepin-devicemanager
- ✅dde-printer
- ✅deepin-graphics-driver-manager
- ✅deepin-turbo
- ✅dde-device-formatter
- ✅deepin-screensaver
- ✅disomaster
- ✅docparser

仍有部分项目未涵盖在此列表，将会根据实际情况补充。若您认为我们可能遗漏了哪个项目，也可以根据通过与我们反馈。

**Reply by kt286** [reply]
dde-grand-search 不是已经迁移了吗？为啥内测的代码没推上来

**Reply by BLumia** [reply]
> dde-grand-search 不是已经迁移了吗？为啥内测的代码没推上来

了解，我明天和对应项目组那边确认下情况

**Reply by BLumia** [reply]
确认了一下，那个项目组确实迁移后还在用 gerrit 进行开发。和那边确认了下问题，下周一会一起确保那边把开发流程迁移过去。另外已确认 dde-grand-search 之前已经从 GitHub 合入的 patch 本身在那边现在所使用的 gerrit 开发分支上并没有丢失，我这边也会在下周一确保相关迁移不会影响到已合入的社区 patch。至于现在的内测版代码（如果你现在就需要的话），应该可以从仓库拿到。

现在还有一些项目还没开始开发流程的迁移，所以暂时也还没有去确认已经迁移但 GitHub 仓库上没有新提交的项目是属于项目本身就没处于活跃开发状态还是走完了迁移流程但还在使用 gerrit。我这边也会在六月迁移计划结束后再确认一下各个项目的实际迁移情况。

**Reply by kt286** [reply]
我是想要提个PR，等迁移过去再提吧

**Reply by BLumia** [reply]
相应[代码已同步](https://github.com/linuxdeepin/dde-grand-search/pull/5)，并且已经和对应迁移步骤有问题的组在核对和解决整个开发流程迁移过程中的问题了。

可以考虑开 PR 了，有其它问题可以直接联系我反馈 :)

**Reply by kt286** [reply]
PR 已经提交 https://github.com/linuxdeepin/dde-grand-search/pull/6

**Comment by BLumia** [comment]
六月的迁移告一段落，其中有三个项目因特殊原因会延迟迁移。后续迁移状态我会更新在这里：

- ✅deepin-editor
- ✅deepin-reader
- ✅deepin-devicemanager

其它项目应当均已迁移，若您发现任何遗漏未迁移的项目，请回复或开启新 issue/讨论贴的形式告知，感谢！

**Reply by utt54321** [reply]
借楼提问：
关于deepin-devicemanager，我发现在github上有相关的库https://github.com/linuxdeepin/deepin-devicemanager
但是其中的changelog只有1.0-1，我使用这个git库clone之后编译，但是会报泛型相关的代码错误
此外，我发现另一个库https://github.com/UbuntuDDE/deepin-devicemanager，然后我使用这个git仓库编译（跳过了测试，在debian11中尝试），可以编译成功，但是会出现segfault导致dbus服务无法正常工作，我想可能原因是在Readme.md中提到的需要polkit的支持，但是对于其提到的
authenticateProxy/policy/com.deepin.pkexec.deepin-devicemanager-authenticateProxy.policy
和
build-all-unknown-Debug/authenticateProxy/deepin-devicemanager-authenticateProx
我完全没有头绪。**是否有大佬知道这两个文件在哪里获取**。

后来我又到163的镜像源中下载deepin-devicemanager的源码包，使用dpkg-source -x解开后，里面的control额外增加了两个依赖，分别是deepin-desktop-base，deepin-keyring
当安装这两个依赖时，编译出来的deepin-devicemanager可以正常显示信息
当使用dpkg --purge删除这两个依赖时，在control中去除两个依赖，编译出来的deepin-devicemanager也会出现上述类似的segfault报错，从而dbus服务无法正常工作。
我非常好奇这是为什么，**我想要在不依赖deepin-desktop-base，deepin-keyring的情况下，编译出可以正常使用的deepin-devicemanager**
我在讨论去提了问题https://github.com/linuxdeepin/developer-center/discussions/3297
求大佬指点

**Reply by BLumia** [reply]
这类问题其实不用借楼问，开个新帖子提问就可以了，这样也便于后续其它遇到和你一样问题的朋友搜索到你的问题。我在 #3297 答复 :)

**Comment by BLumia** [comment]
感谢各位一如既往的关注！我们的 GitHub 开发平台迁移计划现在已基本完成了，希望通过这个迁移事项，能使我们的开发过程变得更加公开、透明，也能使社区的贡献者们能够更加容易的，真正切实的参与到 deepin/DDE 的开发或是其它社区活动中来。

这个主题现在将会被取消置顶。如果你仍有与项目开发流程迁移相关的疑问，或者在参与 deepin/DDE 的贡献过程中需要任何帮助，都随时欢迎你以新建 Issue 或 Discussion 的形式，或是在我们的 [deepin-devel](https://www.freelists.org/list/deepin-devel) 邮件列表中发起讨论，我们会很乐意提供帮助 :)

**Comment by wineee** [comment]
这里迁移成功的标准是什么？ 目前 dde-kwin 没有使用 master 分支，tag 打到了没有同步到 github 的分支上， CI 完全是坏的

**Reply by BLumia** [reply]
:joy: 只是迁到 GitHub 了，这个 thread 只是跟进了开发工作流是否还在内网 gerrit 的问题，和研发主干分支是哪个没关系。部分项目还没切换到 master。初期迁移为了避免对项目组研发进度影响过大也确实没做硬性要求，但最近两周也正在做研发主干分支切到 master 的事情了。预计是下周时所有自研项目切完（这个目标之前没公示）。

deepin-kwin 这个还有点特殊，deepin-kwin 那边觉得 kwin API 太频繁变化，于是打算彻底 fork 出来维护。具体的事项我建议在单独 discussion 里讨论，这个和开发工作流的位置的原始主题偏离太多了...
